build.gradle.kts

```diff
@@ -79,49 +79,45 @@ dependencies {
     compileOnly("org.apache.commons:commons-lang3:3.3.2")
     compileOnly("org.jetbrains:annotations:23.0.0")
 
-    compileOnly("gg.essential:vigilance-1.8.9-forge:+") {
+    compileOnly(libs.vigilance) {
         attributes { attribute(common, true) }
         isTransitive = false
     }
 
-    compileOnly("cc.polyfrost:universalcraft-1.8.9-forge:246") {
+    compileOnly("cc.polyfrost:universalcraft-1.8.9-forge:${libs.versions.universalcraft.get()}") {
         attributes { attribute(common, true) }
         isTransitive = false
     }
 
-    shadeRelocated("com.github.KevinPriv:keventbus:c52e0a2ea0") {
+    shadeRelocated(libs.keventbus) {
         isTransitive = false
     }
 
-    shadeRelocated("com.github.xtrm-en:deencapsulation:42b829f373") {
+    shadeRelocated(libs.deencapsulation) {
         isTransitive = false
     }
 
     @Suppress("GradlePackageUpdate")
-    shadeRelocated("com.github.ben-manes.caffeine:caffeine:2.9.3")
+    shadeRelocated(libs.caffeine)
 
     // for other mods and universalcraft
-    val kotlinVersion: String by project
-    val coroutinesVersion: String by project
-    val serializationVersion: String by project
-    val atomicfuVersion: String by project
-    shade("org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion")
-    shade("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion")
-    shade("org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlinVersion")
-    shade("org.jetbrains.kotlin:kotlin-reflect:$kotlinVersion")
-
-    shade("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion")
-    shade("org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:$coroutinesVersion")
-    shade("org.jetbrains.kotlinx:kotlinx-coroutines-jdk8:$coroutinesVersion")
-    shade("org.jetbrains.kotlinx:kotlinx-serialization-core-jvm:$serializationVersion")
-    shade("org.jetbrains.kotlinx:kotlinx-serialization-json-jvm:$serializationVersion")
-    shade("org.jetbrains.kotlinx:kotlinx-serialization-cbor-jvm:$serializationVersion")
-    shade("org.jetbrains.kotlinx:atomicfu-jvm:$atomicfuVersion")
-
-    shade("org.spongepowered:mixin:0.7.11-SNAPSHOT") {
+    shade(libs.kotlinStdlib)
+    shade(libs.kotlinStdlibJdk8)
+    shade(libs.kotlinStdlibJdk7)
+    shade(libs.kotlinReflect)
+
+    shade(libs.kotlinxCoroutinesCore)
+    shade(libs.kotlinxCoroutinesCoreJvm)
+    shade(libs.kotlinxCoroutinesJdk8)
+    shade(libs.kotlinxSerializationCore)
+    shade(libs.kotlinxSerializationJson)
+    shade(libs.kotlinxSerializationCbor)
+    shade(libs.kotlinxAtomicfu)
```

Comment:
Please use dependency bundles like `libs.bundle.kotlin` ([cool guide](https://proandroiddev.com/gradle-version-catalogs-for-an-awesome-dependency-management-f2ba700ff894))